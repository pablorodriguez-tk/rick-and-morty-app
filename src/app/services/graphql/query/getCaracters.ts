import { gql } from '@apollo/client';

export const GET_CHARACTERS = gql`
  query characters($page: Int, $filter: FilterCharacter) {
    characters(page: $page, filter: $filter) {
      info {
        count
        next
        pages
        prev
      }
      results {
        id
        name
        status
        species
        image
      }
    }
  }
`;
