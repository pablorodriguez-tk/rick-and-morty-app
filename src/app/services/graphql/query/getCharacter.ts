import { gql } from '@apollo/client';

export const GET_CHARACTER = gql`
  query character($characterId: ID!) {
    character(id: $characterId) {
      id
      episode {
        id
        air_date
        name
        episode
      }
    }
  }
`;
