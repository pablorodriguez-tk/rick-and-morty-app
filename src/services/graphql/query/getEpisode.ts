import { gql } from '@apollo/client';

export const GET_EPISODE = gql`
  query episode($episodeId: ID!) {
    episode(id: $episodeId) {
      name
      episode
      characters {
        name
        image
      }
    }
  }
`;
