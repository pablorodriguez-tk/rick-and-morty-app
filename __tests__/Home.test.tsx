import { render, screen, waitFor } from '@testing-library/react';
import Home from '@/app/page';
import { MockedProvider } from '@apollo/client/testing';
import { GET_CHARACTERS } from '@/services/graphql/query/getCaracters';

export const mocks = [
  {
    request: {
      query: GET_CHARACTERS,
      variables: {
        page: 1,
        filter: { name: '' },
      },
    },
    data: {
      characters: {
        info: {
          count: 826,
          next: 2,
          pages: 42,
          prev: null,
        },
        results: [
          {
            id: '1',
            name: 'Rick Sanchez',
            status: 'Alive',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
          },
          {
            id: '2',
            name: 'Morty Smith',
            status: 'Alive',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
          },
          {
            id: '3',
            name: 'Summer Smith',
            status: 'Alive',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/3.jpeg',
          },
          {
            id: '4',
            name: 'Beth Smith',
            status: 'Alive',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/4.jpeg',
          },
          {
            id: '5',
            name: 'Jerry Smith',
            status: 'Alive',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/5.jpeg',
          },
          {
            id: '6',
            name: 'Abadango Cluster Princess',
            status: 'Alive',
            species: 'Alien',
            image: 'https://rickandmortyapi.com/api/character/avatar/6.jpeg',
          },
          {
            id: '7',
            name: 'Abradolf Lincler',
            status: 'unknown',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/7.jpeg',
          },
          {
            id: '8',
            name: 'Adjudicator Rick',
            status: 'Dead',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/8.jpeg',
          },
          {
            id: '9',
            name: 'Agency Director',
            status: 'Dead',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/9.jpeg',
          },
          {
            id: '10',
            name: 'Alan Rails',
            status: 'Dead',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/10.jpeg',
          },
          {
            id: '11',
            name: 'Albert Einstein',
            status: 'Dead',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/11.jpeg',
          },
          {
            id: '12',
            name: 'Alexander',
            status: 'Dead',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/12.jpeg',
          },
          {
            id: '13',
            name: 'Alien Googah',
            status: 'unknown',
            species: 'Alien',
            image: 'https://rickandmortyapi.com/api/character/avatar/13.jpeg',
          },
          {
            id: '14',
            name: 'Alien Morty',
            status: 'unknown',
            species: 'Alien',
            image: 'https://rickandmortyapi.com/api/character/avatar/14.jpeg',
          },
          {
            id: '15',
            name: 'Alien Rick',
            status: 'unknown',
            species: 'Alien',
            image: 'https://rickandmortyapi.com/api/character/avatar/15.jpeg',
          },
          {
            id: '16',
            name: 'Amish Cyborg',
            status: 'Dead',
            species: 'Alien',
            image: 'https://rickandmortyapi.com/api/character/avatar/16.jpeg',
          },
          {
            id: '17',
            name: 'Annie',
            status: 'Alive',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/17.jpeg',
          },
          {
            id: '18',
            name: 'Antenna Morty',
            status: 'Alive',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/18.jpeg',
          },
          {
            id: '19',
            name: 'Antenna Rick',
            status: 'unknown',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/19.jpeg',
          },
          {
            id: '20',
            name: 'Ants in my Eyes Johnson',
            status: 'unknown',
            species: 'Human',
            image: 'https://rickandmortyapi.com/api/character/avatar/20.jpeg',
          },
        ],
      },
    },
  },
];

describe('Home', () => {
  it('Should have text "Character #1"', async () => {
    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Home />
      </MockedProvider>,
    );
    expect(await screen.findByText('Character #1')).toBeInTheDocument();
  });

  it('Should have text "Character #2"', async () => {
    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Home />
      </MockedProvider>,
    );
    expect(await screen.findByText('Character #2')).toBeInTheDocument();
  });

  it('Home Snapshot', async () => {
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Home />
      </MockedProvider>,
    );

    await waitFor(() => {
      expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
