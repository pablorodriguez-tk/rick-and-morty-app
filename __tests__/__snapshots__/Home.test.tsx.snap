// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home Snapshot 1`] = `
<div>
  <main
    class="h-full"
  >
    <div
      class="flex flex-row h-2/3"
    >
      <div
        class="flex flex-col w-1/2 p-4"
      >
        <h1
          class="text-left mb-4 text-3xl font-extrabold text-gray-900 dark:text-white md:text-3xl lg:text-4xl"
        >
          <span
            class="text-transparent bg-clip-text bg-gradient-to-r to-emerald-600 from-sky-400"
          >
            Character #1
          </span>
        </h1>
        <form
          class="flex flex-col justify-center gap-2 w-full lg:flex-row"
        >
          <div
            class="relative w-full"
          >
            <input
              class="block px-2.5 pb-2.5 pt-4 w-full text-sm text-gray-900 bg-transparent rounded-lg border-1 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-blue-500 focus:outline-none focus:ring-0 focus:border-blue-600 peer"
              name="name"
              placeholder=" "
              type="text"
              value=""
            />
            <label
              class="absolute text-sm text-gray-500 dark:text-gray-400 duration-300 transform -translate-y-4 scale-75 top-2 z-10 origin-[0] bg-white dark:bg-gray-900 px-2 peer-focus:px-2 peer-focus:text-blue-600 peer-focus:dark:text-blue-500 peer-placeholder-shown:scale-100 peer-placeholder-shown:-translate-y-1/2 peer-placeholder-shown:top-1/2 peer-focus:top-2 peer-focus:scale-75 peer-focus:-translate-y-4 left-1"
            >
              Name
            </label>
          </div>
        </form>
        <div
          class="grid 2xl:grid-cols-3 xl:grid-cols-2 grid-cols-1 overflow-auto m-2 gap-1"
        />
        <div
          class="flex items-center justify-center text-center"
        >
          <nav
            class=""
          >
            <div
              class="text-sm text-gray-700 dark:text-gray-400"
            >
              Showing 
              <span
                class="font-semibold text-gray-900 dark:text-white"
              >
                1
              </span>
               to 
              <span
                class="font-semibold text-gray-900 dark:text-white"
              >
                0
              </span>
               of 
              <span
                class="font-semibold text-gray-900 dark:text-white"
              >
                0
              </span>
               Entries
            </div>
            <ul
              class="xs:mt-0 mt-2 inline-flex items-center -space-x-px"
            >
              <li>
                <button
                  class="opacity-50 cursor-normal ml-0 rounded-l-lg border border-gray-300 bg-white py-2 px-3 leading-tight text-gray-500 enabled:hover:bg-gray-100 enabled:hover:text-gray-700 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 enabled:dark:hover:bg-gray-700 enabled:dark:hover:text-white"
                  disabled=""
                  type="button"
                >
                  Previous
                </button>
              </li>
              <li>
                <button
                  class="rounded-r-lg border border-gray-300 bg-white py-2 px-3 leading-tight text-gray-500 enabled:hover:bg-gray-100 enabled:hover:text-gray-700 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 enabled:dark:hover:bg-gray-700 enabled:dark:hover:text-white"
                  type="button"
                >
                  Next
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
      <div
        class="flex flex-col w-1/2 p-4"
      >
        <h1
          class="text-right mb-4 text-3xl font-extrabold text-gray-900 dark:text-white md:text-3xl lg:text-4xl"
        >
          <span
            class="text-transparent bg-clip-text bg-gradient-to-r to-emerald-600 from-sky-400"
          >
            Character #2
          </span>
        </h1>
        <form
          class="flex flex-col justify-center gap-2 w-full lg:flex-row"
        >
          <div
            class="relative w-full"
          >
            <input
              class="block px-2.5 pb-2.5 pt-4 w-full text-sm text-gray-900 bg-transparent rounded-lg border-1 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-blue-500 focus:outline-none focus:ring-0 focus:border-blue-600 peer"
              name="name"
              placeholder=" "
              type="text"
              value=""
            />
            <label
              class="absolute text-sm text-gray-500 dark:text-gray-400 duration-300 transform -translate-y-4 scale-75 top-2 z-10 origin-[0] bg-white dark:bg-gray-900 px-2 peer-focus:px-2 peer-focus:text-blue-600 peer-focus:dark:text-blue-500 peer-placeholder-shown:scale-100 peer-placeholder-shown:-translate-y-1/2 peer-placeholder-shown:top-1/2 peer-focus:top-2 peer-focus:scale-75 peer-focus:-translate-y-4 left-1"
            >
              Name
            </label>
          </div>
        </form>
        <div
          class="grid 2xl:grid-cols-3 xl:grid-cols-2 grid-cols-1 overflow-auto m-2 gap-1"
        />
        <div
          class="flex items-center justify-center text-center"
        >
          <nav
            class=""
          >
            <div
              class="text-sm text-gray-700 dark:text-gray-400"
            >
              Showing 
              <span
                class="font-semibold text-gray-900 dark:text-white"
              >
                1
              </span>
               to 
              <span
                class="font-semibold text-gray-900 dark:text-white"
              >
                0
              </span>
               of 
              <span
                class="font-semibold text-gray-900 dark:text-white"
              >
                0
              </span>
               Entries
            </div>
            <ul
              class="xs:mt-0 mt-2 inline-flex items-center -space-x-px"
            >
              <li>
                <button
                  class="opacity-50 cursor-normal ml-0 rounded-l-lg border border-gray-300 bg-white py-2 px-3 leading-tight text-gray-500 enabled:hover:bg-gray-100 enabled:hover:text-gray-700 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 enabled:dark:hover:bg-gray-700 enabled:dark:hover:text-white"
                  disabled=""
                  type="button"
                >
                  Previous
                </button>
              </li>
              <li>
                <button
                  class="rounded-r-lg border border-gray-300 bg-white py-2 px-3 leading-tight text-gray-500 enabled:hover:bg-gray-100 enabled:hover:text-gray-700 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 enabled:dark:hover:bg-gray-700 enabled:dark:hover:text-white"
                  type="button"
                >
                  Next
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
    <div
      class=" fixed right-6 bottom-6 group"
      data-dial-init="true"
    >
      <button
        aria-controls="speed-dial-menu-default"
        aria-expanded="false"
        class="group p-0.5 text-center font-medium relative focus:z-10 bg-cyan-700 border-transparent enabled:hover:bg-cyan-800 dark:bg-cyan-600 dark:enabled:hover:bg-cyan-700 border-0 text-gray-900 rounded-lg group bg-gradient-to-br from-green-400 to-blue-600 group-hover:from-green-400 group-hover:to-blue-600 hover:text-white dark:text-white focus:ring-4 focus:outline-none focus:ring-green-200 dark:focus:ring-green-800 flex items-center justify-center"
        data-dial-toggle="speed-dial-menu-default"
        type="button"
      >
        <span
          class="items-center flex justify-center bg-white text-gray-900 transition-all duration-75 ease-in group-enabled:group-hover:bg-opacity-0 group-enabled:group-hover:text-inherit dark:bg-gray-900 dark:text-white w-full rounded-md text-sm px-4 py-2 border border-transparent"
        >
          <svg
            class="w-5 h-5 transition-transform group-hover:scale-150"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 10l7-7m0 0l7 7m-7-7v18"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span
            class="sr-only"
          >
            Open actions menu
          </span>
        </span>
      </button>
    </div>
    <button
      class="group flex items-center justify-center p-0.5 text-center font-medium focus:z-10 focus:outline-none text-white bg-gradient-to-br from-green-400 to-cyan-600 enabled:hover:bg-gradient-to-bl focus:ring-green-200 dark:focus:ring-green-800 border-0 rounded-lg focus:ring-2 absolute top-4 left-1/2 transform -translate-x-1/2"
      type="button"
    >
      <span
        class="items-center flex justify-center bg-white text-gray-900 transition-all duration-75 ease-in group-enabled:group-hover:bg-opacity-0 group-enabled:group-hover:text-inherit dark:bg-gray-900 dark:text-white w-full rounded-md text-sm px-4 py-2 border border-transparent"
      >
        <p>
          Reset
        </p>
      </span>
    </button>
  </main>
</div>
`;
